compiler/rustc_codegen_llvm/src/debuginfo/dwarf_const.rs

```diff
@@ -0,0 +1,28 @@
+//! Definitions of various DWARF-related constants.
+//! These should agree with `llvm/include/llvm/BinaryFormat/Dwarf.def`.
+
+#![allow(non_upper_case_globals)] // Allow names to match their LLVM equivalents
+
+use libc::c_uint;
+
+pub(crate) const DW_TAG_const_type: c_uint = 0x26;
+
+/// DWARF constant representing the Rust language. Introduced in DWARF 5.
+/// Defined by <http://www.dwarfstd.org/ShowIssue.php?issue=140129.1>.
+pub(crate) const DW_LANG_RUST: c_uint = 0x1c;
+
+// DWARF attribute type encodings
+
+pub(crate) const DW_ATE_boolean: c_uint = 0x02;
+pub(crate) const DW_ATE_float: c_uint = 0x04;
+pub(crate) const DW_ATE_signed: c_uint = 0x05;
+pub(crate) const DW_ATE_unsigned: c_uint = 0x07;
+pub(crate) const DW_ATE_UTF: c_uint = 0x10;
+
+// DWARF expression operators
+
+pub(crate) const DW_OP_deref: u64 = 0x06;
+pub(crate) const DW_OP_plus_uconst: u64 = 0x23;
+/// Defined by LLVM in `llvm/include/llvm/BinaryFormat/Dwarf.h`.
+/// Double-checked by a static assertion in `RustWrapper.cpp`.
+pub(crate) const DW_OP_LLVM_fragment: u64 = 0x1000;
```

Comment:
You can also import them from gimli, right? We already depend on gimli anyway for both the backtrace and thorin crates.

Reply:
Oh, that's a good idea.

We still have to define `DW_OP_LLVM_fragment` ourselves, but all the standard ones should be in gimli already.

Reply:
I'm surprised that `DW_OP_LLVM_fragment` isn't in gimli given that it already has both gnu and wasm extensions supported.

Reply:
I don't know the full details, but I think `DW_OP_LLVM_fragment` might be used internally by LLVM as a sentinel value, without appearing in the output? I could be wrong though.

Reply:
(Note that even LLVM doesn't define that value in its `Dwarf.def` that holds all the other DWARF constants; it gets defined separately in a regular header.)